File: 1207. Unique Number of Occurrences/unique_occurrences.rs
use std::collections::{HashMap, HashSet};

impl Solution {
    
    // O(n) solution,
    // O(n) space,
    // Approach: hashtable, hashset
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        
        let mut nums_count: HashMap<i32, i32> = HashMap::new();
        
        for num in arr {
            
            let mut count = 0;
            
            if nums_count.contains_key(&num) {
                count = *nums_count.get(&num).unwrap_or(&0);
            }
            
            count += 1;
            
            nums_count.insert(num, count);
        }
        
        let mut frequencies: HashSet<i32> = HashSet::new();
        
        for (num, count) in nums_count {
            if frequencies.contains(&count) {
                return false;
            }
            frequencies.insert(count);
        }
        
        return true;
    }
}

File: 1207. Unique Number of Occurrences/UniqueOccurrences.py
from collections import Counter
from typing import List


class Solution:
    
    # O(n) time,
    # O(n) space,
    # Approach: couting, hash map
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occ_count = Counter(Counter(arr).values())

        for val in occ_count.values():
            if val > 1:
                return False
            
        return True